api/model/api.go

```diff
@@ -508,6 +508,16 @@ type OpenIDOptions struct {
 	SegregateByClient bool                `json:"segregate_by_client"`
 }
 
+// PinnedPublicKeySecret holds information about a Kubernetes Secret object called 'SecretName' residing in
+// 'SecretNamespace' namespace that includes the public key of the domain, defined through 'PublicKeySecretField' field.
+type PinnedPublicKeySecret struct {
+	SecretName string `json:"secret_name"`
+
+	// +optional
+	// +kubebuilder:default:=default
+	SecretNamespace string `json:"secret_namespace,omitempty"`
```

Comment:
The namespace should be the same as the one used in the ApiDefinition => the check should be done there.